src/middlewared/middlewared/plugins/directoryservices_/ipa_join_mixin.py

```diff
@@ -48,21 +47,13 @@ def _parse_ipa_response(resp: subprocess.CompletedProcess) -> dict:
 class IPAJoinMixin:
     __ipa_smb_domain = undefined
 
-    def _ipa_remove_kerberos_cert_config(self, job: Job | None, ldap_config: dict | None):
-        if ldap_config is None:
-            ldap_config = self.middleware.call_sync('ldap.config')
-
-        if job:
-            job.set_progress(80, 'Removing kerberos configuration.')
-
-        if ldap_config['kerberos_realm']:
-            # TODO: remove after unifying the directory services plugins
-            # User may have configured our kerberos realm in AD plugin
-            # necessitating deletion from config
-            self.middleware.call_sync(
-                'datastore.update', 'directoryservice.activedirectory', 1, {'ad_kerberos_realm': None}
-            )
-            self.middleware.call_sync('kerberos.realm.delete', ldap_config['kerberos_realm'])
+    def _ipa_remove_kerberos_cert_config(self, job: Job | None, ds_config: dict | None):
+        if ds_config['kerberos_realm']:
+            self.middleware.call_sync('datastore.update', 'directoryservices', 1, {
+                'kerberos_realm_id': None,
+            })
+            if realm := self.middleware.call_sync('kerberos.realm.query', [['realm', '=', ds_config['kerberos_realm']]]):
```

Comment:
To keep the line under 120:
```suggestion
            if (realm := self.middleware.call_sync('kerberos.realm.query', [
                ['realm', '=', ds_config['kerberos_realm']]
            ])):
```